File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        look = dict()
        for i, n in enumerate(nums):
            if n in look and (i - look[n] <= k):
                return True
            look[n] = i
        return False

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.ts
function containsNearbyDuplicate(nums: number[], k: number): boolean {
    const lookup = new Map<number, number>();

    for (let i = 0; i < nums.length; i++) {
        const n = nums[i];

        if (lookup.has(n) && (i - lookup.get(n) <= k)) return true;

        lookup.set(n, i);
    }

    return false;
};